### PR TITLE
chore: assert_contains_ignore_case

### PR DESCRIPTION
**Pull request by raphaelstolt**
## 📚 Description

Adds the `assert_contains_ignore_case` assertion.

## ✅ To-do list

- [x] I updated the `CHANGELOG.md` to reflect the new feature
- [x] I updated the documentation to reflect the changes

### CODE REVIEW

**Comment by Chemaclass** [comment]
Thanks for contributing, @raphaelstolt! Can you double-check the [failing tests](https://github.com/TypedDevs/bashunit/actions/runs/6534911497/job/17743402183?pr=196)?

Screenshot 2023-10-16 at 16 25 52

**Review by Chemaclass — APPROVED** [review]
Thanks a lot! 👏

File: docs/assertions.md [hunk 1]
```diff
@@ -40,6 +40,15 @@ function test_failure() {
...
+
+:::
+
+## assert_contains_ignore_case
+> `assert_contains_ignore_case "needle" "haystack"`
+
+Reports an error if `needle` is not a substring of `haystack`. Differences in casing are ignored when needle is searched
+for in haystack.
+
 :::
```
Comment (Tito-Kati) [inline docs/assertions.md:52]:
````suggestion
## assert_contains_ignore_case
> `assert_contains_ignore_case "needle" "haystack"`

Reports an error if `needle` is not a substring of `haystack`.
Differences in casing are ignored when needle is searched for in haystack.

::: code-group
```bash [Example]
function test_success() {
  assert_contains_ignore_case "foo" "FooBar"
}

function test_failure() {
  assert_contains_ignore_case "baz" "FooBar"
}
```
:::
````

**Review by Tito-Kati — CHANGES_REQUESTED** [review]
Thank you so much for your contribution @raphaelstolt ! If you don't mind adding my suggestion (a good example in the documentation for a better DX) it would be the icing on the cake.

**Comment by Tito-Kati** [comment]
closes #178

**Review by JesusValera — APPROVED** [review]

**Comment by raphaelstolt** [comment]
Documentation is updated as suggested.

**Review by khru — APPROVED** [review]
Looks good

File: docs/assertions.md (hunk 1 above)
Comment (khru) [reply docs/assertions.md]: Done @Tito-Kati

**Review by Tito-Kati — APPROVED** [review]

**Comment by khru** [comment]
Kudo @raphaelstolt you've done a nice job, thank you for helping and supporting the project :heart:

**Pull request merged**